contracts/instruments/RibbonThetaVaultYearn.sol

```diff
@@ -0,0 +1,720 @@
+        newAssetBalance = total.sub(withdrawAmount);
+        newShareSupply = shareSupply.sub(share);
+    }
+
+    /**
+     * @notice Returns the max withdrawable shares for all users in the vault
+     */
+    function maxWithdrawableShares() public view returns (uint256) {
+        uint256 withdrawableBalance = assetBalance();
+        uint256 total = lockedAmount.add(withdrawableBalance);
+        return
+            withdrawableBalance
+                .mul(totalSupply())
+                .div(total)
+                .sub(MINIMUM_SUPPLY)
+                .sub(queuedWithdrawShares);
+    }
+
+    /**
+     * @notice Returns the max amount withdrawable by an account using the account's vault share balance
+     * @param account is the address of the vault share holder
+     * @return amount of `asset` withdrawable from vault, with fees accounted
+     */
+    function maxWithdrawAmount(address account)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 maxShares = maxWithdrawableShares();
+        uint256 share = balanceOf(account);
+        uint256 numShares = min(maxShares, share);
+
+        (uint256 withdrawAmount, , ) =
+            _withdrawAmountWithShares(numShares, assetBalance());
+
+        return withdrawAmount;
+    }
+
+    /**
+     * @notice Returns the number of shares for a given `assetAmount`.
+     *         Used by the frontend to calculate withdraw amounts.
+     * @param assetAmount is the asset amount to be withdrawn
+     * @return share amount
+     */
+    function assetAmountToShares(uint256 assetAmount)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 total = lockedAmount.add(assetBalance());
+        return assetAmount.mul(totalSupply()).div(total);
+    }
+
+    /**
+     * @notice Returns an account's balance on the vault
+     * @param account is the address of the user
+     * @return vault balance of the user
+     */
+    function accountVaultBalance(address account)
+        external
+        view
+        returns (uint256)
+    {
+        (uint256 withdrawAmount, , ) =
+            _withdrawAmountWithShares(balanceOf(account), assetBalance());
+        return withdrawAmount;
+    }
+
+    /**
+     * @notice Returns the vault's total balance, including the amounts locked into a short position
+     * @return total balance of the vault, including the amounts locked in third party protocols
+     */
+    function totalBalance() public view returns (uint256) {
+        return lockedAmount.add(assetBalance());
+    }
+
+    /**
+     * @notice Returns the asset balance on the vault. This balance is freely withdrawable by users.
+     */
+    function assetBalance() public view returns (uint256) {
+        return
+            IERC20(asset).balanceOf(address(this)).add(
+                IERC20(address(collateralToken)).balanceOf(address(this)).mul(
+                    collateralToken.pricePerShare()
+                )
```

Comment:
Does this account for withdrawal slippage?

Reply:
addressed